GitUI/CommandsDialogs/SettingsDialog/Plugins/PluginSettingsPage.cs

```diff
@@ -59,14 +60,14 @@ private IEnumerable<ISetting> GetSettings()
                 throw new ApplicationException();
             }
 
-            return _gitPlugin.HasSettings ? _gitPlugin.GetSettings() : Array.Empty<ISetting>();
+            return _gitPlugin.GetSettings() ?? Array.Empty<ISetting>();
         }
 
         public override SettingsPageReference PageReference => new SettingsPageReferenceByType(_gitPlugin.GetType());
 
         protected override ISettingsLayout CreateSettingsLayout()
         {
-            labelNoSettings.Visible = !_gitPlugin.HasSettings;
+            labelNoSettings.Visible = !_gitPlugin.GetSettings()?.Any() ?? false;
```

Comment:
Loading settings maybe an expensive exercise (e.g. https://github.com/gitextensions/gitextensions/issues/8339), this can cause an observable perf regressions. 
This is one of reasons why we added a flag - a plugin author is responsible for indicating whether a plugin has settings or not.

Reply:
Last code is

`labelNoSettings.Visible = !(_gitPlugin is IGitPluginConfigurable);`